tests-e2e/cypress/integration/runs/rdp_main_header_spec.js

```diff
@@ -0,0 +1,494 @@
+                        const message = 'Status update - ' + Date.now();
+                        cy.apiUpdateStatus({
+                            playbookRunId: playbookRun.id,
+                            message,
+                        });
+
+                        // # Navigate to the town square channel
+                        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+                        // * Verify that the last post does not contain the status update
+                        cy.getLastPost().then((post) => {
+                            cy.get(post).contains(message).should('not.exist');
+                        });
+
+                        // # Navigate to the off-topic channel
+                        cy.visit(`/${testTeam.name}/channels/off-topic`);
+
+                        // * Verify that the last post does not contain the status update
+                        cy.getLastPost().then((post) => {
+                            cy.get(post).contains(message).should('not.exist');
+                        });
+                    });
+                });
+            });
+        });
+
+        describe('context menu', () => {
+            commonContextDropdownTests();
+
+            describe('finish run', () => {
+                it('can be confirmed', () => {
+                    // # Open dropdown
+                    cy.findByTestId('run-header-section').find('h1').click();
+
+                    // # Click on finish run
+                    getDropdownItemByText('Finish run').click();
+
+                    // * Check that status badge is in-progress
+                    cy.findByTestId('run-header-section').findByTestId('badge').contains('In Progress');
+
+                    // # Check that finish run modal is open
+                    cy.get('#confirmModal').should('be.visible');
+                    cy.get('#confirmModal').find('h1').contains('Confirm finish run');
+
+                    // # Click on confirm
+                    cy.get('#confirmModal').get('#confirmModalButton').click();
+
+                    // # Open dropdown
+                    cy.findByTestId('run-header-section').find('h1').click();
+
+                    // * Assert option is not anymore in context dropdown
+                    getDropdownItemByText('Finish run').should('not.exist');
+
+                    // * Assert status badge is finished
+                    cy.findByTestId('run-header-section').findByTestId('badge').contains('Finished');
+                });
+
+                it('can be canceled', () => {
+                    // # Open dropdown
+                    cy.findByTestId('run-header-section').find('h1').click();
+
+                    // # Click on finish run
+                    getDropdownItemByText('Finish run').click();
+
+                    // * Check that status badge is in-progress
+                    cy.findByTestId('run-header-section').findByTestId('badge').contains('In Progress');
```

Comment:
Same comment here about swapping these two steps
```suggestion
                    // * Check that status badge is in-progress
                    cy.findByTestId('run-header-section').findByTestId('badge').contains('In Progress');

                    // # Click on finish run
                    getDropdownItemByText('Finish run').click();
```